src/v5424.rs

```diff
@@ -238,7 +293,45 @@ const VERSION: &str = "1";
 /// application is incapable of obtaining system time.
 ///
 /// [spec](https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.3)
-type Timestamp = String;
+pub enum Timestamp<'a> {
+    /// Provide a datatime to be formatted.
```

Comment:
Is there any _benefit_ for the caller to provide _their_ chrono local timestamp iso let the formatter create one? Maybe one could live with this _option_.

Reply:
Well they may have already created one, for example if you're logging to multiple targets you can create a single datetime and give it to many formatters.

It's actually handier to take a reference to a DateTime :thinking:

Reply:
Resolved in https://github.com/tandemdrive/syslog-fmt/pull/18/commits/99af8584050d0a5af2d6dad425e53324dcd399e1

Reply:
:+1: